base/unicode/properties.jl

```diff
@@ -0,0 +1,219 @@
+const global Sm = 19  # Symbol, math
+const global Sc = 20  # Symbol, currency
+const global Sk = 21  # Symbol, modifier
+const global So = 22  # Symbol, other
+const global Zs = 23  # Separator, space
+const global Zl = 24  # Separator, line
+const global Zp = 25  # Separator, paragraph
+const global Cc = 26  # Other, control
+const global Cf = 27  # Other, format
+const global Cs = 28  # Other, surrogate
+const global Co = 29  # Other, private use
+
+abstract UnicodeProperty
+
+abstract CharCategory   <: UnicodeProperty
+
+abstract CatLetter      <: CharCategory
+abstract CatMark        <: CharCategory
+abstract CatNumber      <: CharCategory
+abstract CatPunctuation <: CharCategory
+abstract CatSymbol      <: CharCategory
+abstract CatSeparator   <: CharCategory
+abstract CatOther       <: CharCategory
+
+abstract CatUpper       <: CatLetter
+
+abstract LowerCase      <: CatLetter
+abstract UpperCase      <: CatUpper
+abstract TitleCase      <: CatUpper
+abstract ModifierLetter <: CatLetter
+abstract OtherLetter    <: CatLetter
+
+abstract NonSpacingMark <: CatMark
+abstract CombiningMark  <: CatMark
+abstract EnclosingMark  <: CatMark
+
+abstract DecimalDigit   <: CatNumber
+abstract NumericLetter  <: CatNumber
+abstract OtherNumber    <: CatNumber
+
+abstract ConnectorPunct <: CatPunctuation
+abstract DashPunct      <: CatPunctuation
+abstract OpenPunct      <: CatPunctuation
+abstract ClosePunct     <: CatPunctuation
+abstract BegQuotePunct  <: CatPunctuation
+abstract EndQuotePunct  <: CatPunctuation
+abstract OtherPunct     <: CatPunctuation
+
+abstract MathSymbol     <: CatSymbol
+abstract CurrencySymbol <: CatSymbol
+abstract ModifierSymbol <: CatSymbol
+abstract OtherSymbol    <: CatSymbol
+
+abstract SpaceSeparator <: CatSeparator
+abstract LineSeparator  <: CatSeparator
+abstract ParagraphSeparator <: CatSeparator
+
+abstract NotAssignedChar <: CatOther
+abstract ControlChar     <: CatOther
+abstract FormatChar      <: CatOther
+abstract SurrogateChar   <: CatOther
+abstract PrivateUseChar  <: CatOther
+
+const c2t = [NotAssignedChar, UpperCase, LowerCase, TitleCase, ModifierLetter, OtherLetter,
+             NonSpacingMark, CombiningMark, EnclosingMark,
+             DecimalDigit, NumericLetter, OtherNumber,
+             ConnectorPunct, DashPunct, OpenPunct, ClosePunct,
+             BegQuotePunct, EndQuotePunct, OtherPunct,
+             MathSymbol, CurrencySymbol, ModifierSymbol, OtherSymbol,
+             SpaceSeparator, LineSeparator, ParagraphSeparator,
+	     ControlChar, FormatChar, SurrogateChar, PrivateUseChar]
+
+############################################################################
+
+
+"""
+Return different Unicode properties for character
+"""
+function charprop end
+
+charprop(::Type{CharCategory}, c) = c2t[charprop(CharCategoryCode, c)+1]
+
+is_assigned_char(c) = charprop(CharCategoryCode, c) != Cn
+
+## libc character class predicates ##
+
+islower(c::Char) = (charprop(CharCategoryCode, c) == Ll)
```

Comment:
That was like that in the old code, and I in fact prefer to have the parenthesis, even though they are unnecessary in this case in Julia (too much time dealing with different languages that have different precedence levels, so my rule is to use parenthesis for clarity, even when not necessary in any one particular language).